lib/network/probe/ProbeServer.js

```diff
@@ -0,0 +1,109 @@
+const httpServer = require('../http/server');
+const werelogs = require('werelogs');
+const errors = require('../../errors');
+
+const DEFAULT_LIVE_ROUTE = '/_/live';
+const DEFAULT_READY_ROUTE = '/_/live';
+const DEFAULT_METRICS_ROUTE = '/_/metrics';
+
+/**
+ * ProbeDelegate is used to determine if a probe is successful or
+ * if any errors are present.
+ * If everything is working as intended, it is a no-op.
+ * Otherwise, return a string representing what is failing.
+ * @callback ProbeDelegate
```

Comment:
How is ProbeDelegate a callback?

Reply:
There is not jsdoc for delegates so I used callback here as the closest but also included the @returns to specify you can return values like a normal delegate.

Reply:
https://jsdoc.app/tags-callback.html says you can specify the return value, seems JsDoc does not differentiate between a callback ( no return value ) and a delegate ( one or more values ) and just has one group. But we can at least use naming to separate them.

Reply:
This reminds me of our discussion of specifying `@returns {undefined}` or `@returns {callback}`.
I think this info completes that loop :)